src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -1256,6 +1256,13 @@ private void ReturnConnectionToPoolCore()
                 }
             }
 
+            // If we have extraneous data in the read buffer, don't reuse the connection;
+            // otherwise we'd interpret this as part of the next response.
+            if (_readOffset != _readLength)
+            {
+                _connectionClose = true;
+            }
```

Comment:
Why set _connectionClose rather than just adding this to the below if condition?

Reply:
Should we do a similar check earlier and fail the request if we detect this?